lib/Conversion/MooreToCore/MooreToCore.cpp

```diff
@@ -713,6 +713,73 @@ static Value createZeroValue(Type type, Location loc,
   return rewriter.createOrFold<hw::BitcastOp>(loc, type, constZero);
 }
 
+struct ClassPropertyRefOpConversion
+    : public OpConversionPattern<circt::moore::ClassPropertyRefOp> {
+  ClassPropertyRefOpConversion(TypeConverter &tc, MLIRContext *ctx,
+                               ClassTypeCache &cache)
+      : OpConversionPattern(tc, ctx), cache(cache) {}
+
+  LogicalResult
+  matchAndRewrite(circt::moore::ClassPropertyRefOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    MLIRContext *ctx = rewriter.getContext();
+
+    // Convert result type; we expect !llhd.ref<someT>.
+    Type dstTy = getTypeConverter()->convertType(op.getPropertyRef().getType());
+    auto dstRefTy = dyn_cast<llhd::RefType>(dstTy);
+    if (!dstRefTy)
+      return rewriter.notifyMatchFailure(op,
+                                         "result must lower to !llhd.ref<T>");
+    // Operand is a !llvm.ptr
+    Value instRef = adaptor.getInstance();
+    auto instRefTy = dyn_cast<LLVM::LLVMPointerType>(instRef.getType());
+    if (!instRefTy)
+      return rewriter.notifyMatchFailure(op, "instance must be !llvm.ptr");
```

Comment:
This should also be covered by the op verifier. You can probably drop `instRefTy`.

Reply:
Sure, check is gone